java/ql/src/experimental/Security/CWE/CWE-489/ServletMain.ql

```diff
@@ -0,0 +1,63 @@
+/**
+ * @name Main Method in Java EE Web Components
+ * @description Jave EE web applications with a main method.
+ * @kind problem
+ * @id java/main-method-in-web-components
+ * @tags security
+ *       external/cwe-489
+ */
+
+import java
+import semmle.code.java.frameworks.Servlets
+
+/** The java type `javax.servlet.Filter`. */
+class ServletFilterClass extends Class {
+  ServletFilterClass() { this.getASupertype*().hasQualifiedName("javax.servlet", "Filter") }
+}
+
+/** Listener class in the package `javax.servlet` and `javax.servlet.http` */
+class ServletListenerClass extends Class {
+  // Various listener classes of Java EE such as ServletContextListener. They all have a name ending with the word "Listener".
+  ServletListenerClass() {
+    this.getASupertype*()
+        .getQualifiedName()
+        .regexpMatch([
+            "javax\\.servlet\\.[a-zA-Z]+Listener", "javax\\.servlet\\.http\\.[a-zA-Z]+Listener"
+          ])
+  }
+}
+
+/** The `main` method in `Servlet`. */
+class ServletMainMethod extends Method {
```

Comment:
Due to my change requests this is not only a `ServletMainMethod`, but rather a `WebComponentMainMethod`. Could you please update the class and file names to reflect this change?

Reply:
I've renamed the method and the query.